style-guide: refresh and add shell-specific pages (#7441)
tldr-pages/tldr: contributing-guides/style-guide.md

The basic format of each page should match the following template:

```md
# <command-name>

> Short, snappy description.
> Preferably one line; two are acceptable if necessary.
> More information: <https://example.com>.

- Example description:

> `<command-name> -opt1 -opt2 -arg1 {{arg_value}}`

Reviewer: This is probably best left as-is.

- Example description:

> `<command-name> -opt1 -opt2`

Reviewer: Same here.

- In case of a possible reference both to a file or a directory,
  use `{{path/to/file_or_directory}}`.

### Extensions

- If a particular extension is expected for the file, append it.
  For example, `unrar x {{compressed.rar}}`.
- In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
  For instance, in `find.md`'s example "Find files by extension" (`find {{root_path}} -name '{{*.ext}}'`)
  using `{{*.ext}}` explains the command without being unnecessarily specific;
  while in `wc -l {{file}}` using `{{file}}` (without extension) is sufficient.
  

> ### Options
> 
> - Prefer long options instead of short ones whenever is possible.

Reviewer 2: I think they're mostly referred to as "flags".
```suggestion
### Flags

- Prefer long flags instead of short ones whenever it is possible. E.g. use `--command`, instead of `-c`. This makes it easier for the user to remember and understand them.
```

Author: As I know flags are options without value. That's why I used `options` term.

Reviewer 3: Options is definitely not a good fit, if I was new, I would interpret that as being the token syntax at a glance

Reviewer 3: I'd reword this to:

```suggestion
### Flags

- Always use long flags when applicable, for example instead of `bash -c`, use `bash --command`, so the reader can understand the meaning of a flag.
```

- In case of a possible reference both to a file or a directory,
  use `{{path/to/file_or_directory}}`.

### Extensions

- If a particular extension is expected for the file, append it.
  For example, `unrar x {{compressed.rar}}`.
- In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
  For instance, in `find.md`'s example "Find files by extension" (`find {{root_path}} -name '{{*.ext}}'`)
  using `{{*.ext}}` explains the command without being unnecessarily specific;
  while in `wc -l {{file}}` using `{{file}}` (without extension) is sufficient.
  
### Flags

> - Always use long flags when applicable, for example instead of `fish -c`, use `fish --command`, so the reader can understand the meaning of a flag.

Reviewer: This is a convention yes, but it doesn't always hold true. It's actually a case-by case basis. For example, on pages like `ls` or `git commit` which are extremely frequently used we do not actually recommend `--long-arguments`.

- In case of a possible reference both to a file or a directory,
  use `{{path/to/file_or_directory}}`.

### Extensions

- If a particular extension is expected for the file, append it.
  For example, `unrar x {{compressed.rar}}`.
- In case a generic extension is needed, use `{{.ext}}`, but **only** if an extension is required.
  For instance, in `find.md`'s example "Find files by extension" (`find {{root_path}} -name '{{*.ext}}'`)
  using `{{*.ext}}` explains the command without being unnecessarily specific;
  while in `wc -l {{file}}` using `{{file}}` (without extension) is sufficient.
  
### Flags

> - Always use long flags when available for pages in `linux/` directory. For example instead of `fish -c`, use `fish --command`, so the reader can understand the meaning of a flag.

Reviewer 4: ```suggestion
- Use long flags when available for pages in `linux/` directory. For example instead of `fish -c`, use `fish --command`, so the meaning of the flag can be understood.
```

- Extensions, ex. `.dll`.
- Commands, ex. `ls`.

### Shell-specific pages

If a page describes some shell (Bash for instance) then when it's possible try follow this structure:

- Start an interactive shell session:
- Start an interactive shell session without loading startup configs:
- Execute a command:
- Execute a script:
- Check a script for syntax errors:
- _any other examples_
- Print the version:

> If you want to include shell builtin sample please mark it with `(builtin)` word. For example:

Reviewer 3: "What in the world is a shell builtin sample" Me + most others (im guessing)

Even from reading the "For example:" I still have no idea what you are referring to.

Author: Now this sentence sounds like:

> If you want to include shell builtin sample please place `(builtin)` word at the end of example description such as:

```md
# fish

> The Friendly Interactive SHell, a command-line interpreter designed to be user friendly.
> More information: <https://fishshell.com>.

- Start an interactive shell session:

`fish`

- Start an interactive shell session without loading startup configs:

`fish --no-config`

- Execute a command:

`fish --command "{{command}}"`

- Execute a script:

`fish {{path/to/script.fish}}`

- Check a script for syntax errors:

`fish --no-execute {{path/to/script.fish}}`

- Start an interactive shell session in private mode, where the shell does not access old history or save new history:

`fish --private`

- Define and export environmental variable that persist across shell restarts (builtin):

`set --universal --export {{variable_name}} {{variable_value}}`

- Print the version:

`fish --version`
```

> 

Reviewer 2: Thanks for the effort, but I think that's really too much for a general style guide. If it becomes too long it's hard to find the thing you're searching for.

- Extensions, ex. `.dll`.
- Commands, ex. `ls`.

> ### Shell-specific pages

Reviewer: Hrm, I feel like this is overly-specific and isn't a great fit for the style guide. Most pages do not document a shell, so aren't affected by this.

Reviewer 4: Maybe this part could get its own file. There is not that much shells, but there's a lot of builtins.
Even though it's specific, might be good to have this things standardized.

Author: Please view [this](https://github.com/Console-Utils/tldr/blob/shell-guide/contributing-guides/tool-oriented-guides/shells.md) file.

> If a page describes some shell (Bash for instance) then when it's possible try follow this structure:

Reviewer 5: ```suggestion
If a page describes a shell, when it's possible try to follow this structure:
```
